libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -575,12 +609,8 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
     IOT_FUNCTION_ENTRY( IotMqttError_t, IOT_MQTT_SUCCESS );
     _mqttOperation_t * pSubscriptionOperation = NULL;
 
-    /* Subscription serializer function. */
-    IotMqttError_t ( * serializeSubscription )( const IotMqttSubscription_t *,
-                                                size_t,
-                                                uint8_t **,
-                                                size_t *,
-                                                uint16_t * ) = NULL;
+    /* Default to use the appropriate subscribe serializer */
+    _subscribeSerializer_t serializeSubscription = _getMqttSubscribeSerializer( mqttConnection->pSerializer );
```

Comment:
Shouldn't this be called after _checkInit() to make sure mqttConnection was initialized?

Reply:
Good catch, and in fact this was the cause of the unit test failure.  I'm able to run all the tests locally now so will verify this is fixed prior to the next revision